### PR TITLE
Add pyproject.toml and update tox.ini

### PR DESCRIPTION
**Pull request by jnsgruk**
As the description says, introduce a `tox.ini` and a `pyproject.toml` to the project.

Note: linting fails at this stage, fixes needed for it to pass are in #25

### CODE REVIEW

**Review by rbarry82 — APPROVED** [review]
LGTM after an update to make mypy happy.

It definitely doesn't like `lib/charms/...` compared to `import charms.` in `src/charm.py`, though. We should either get :star:craft to redefine the build/deployment strategy, or run mypy once against `tests/` and once against everything else.

I'm surprised this hasn't cropped up in other charms.

**Comment by jnsgruk** [comment]
> LGTM after an update to make mypy happy.
> 
> It definitely doesn't like `lib/charms/...` compared to `import charms.` in `src/charm.py`, though. We should either get craft to redefine the build/deployment strategy, or run mypy once against `tests/` and once against everything else.
> 
> I'm surprised this hasn't cropped up in other charms.

Fixed ;-)

It's because you were importing libraries as `lib.charms.whatever`, you can just import them as `charms.` because of the modified `PYTHONPATH`. With the last commit, the following all pass:

```
tox -e lint
tox -e static
tox -e unit
```

**Review by rbarry82 — APPROVED** [review]

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+# See LICENSE file for licensing details.
+[tox]
+skipsdist=True
+skip_missing_interpreters = True
+envlist = lint, static, unit
+
+[vars]
+src_path = {toxinidir}/src/
+tst_path = {toxinidir}/tests/
+lib_path = {toxinidir}/lib/charms/grafana_k8s/
```
Comment (sed-i) [inline tox.ini:11]: the `/` suffix is redundant?

**Review by sed-i — APPROVED** [review]

**Comment by jnsgruk** [comment]
Merge whenever you're ready @rbarry82 :)

**Pull request merged**